lib/memory.h

```diff
@@ -142,6 +142,16 @@ struct memgroup {
 	}                                                                      \
 	MACRO_REQUIRE_SEMICOLON() /* end */
 
+#ifdef __INTELLISENSE__
+/*
+ * Fix Visual Studio Code error: attribute "constructor" does not take
```

Comment:
This is also used in other places. There's a definition for `_CONSTRUCTOR` in lib/compiler.h:

https://github.com/FRRouting/frr/blob/79e9f21287c9164b38324324a35ae65cfccd600e/lib/compiler.h#L39-L124

(and - related - legacy/historic Solaris had no support for parameters on `constructor` either, which is handled there)

=> move this to `lib/compiler.h` next to `_CONSTRUCTOR` handling